Respect KeytabPrincipal setting in VerifyAPREQ

tyrannosaurus-becks opened the pull request:
Fixes #330 

This PR allows a caller to override the keytab principal used in `VerifyAPREQ`. Since this method is used by the `SPNEGOKRB5Authenticate` function, this fixes the issue in the linked ticket where the given keytab principal was being ignored.

Please let me know if this is not the right approach, and a thank-you in advance to the maintainers of this repo!

jcmturner commented:
Fixes #275

jcmturner commented:
Hi @tyrannosaurus-becks thanks for this contribution. I think your solution is the right one. However this is a breaking change so needs to be made on v8 rather than v7. As part of v8 I have done some work to adopt Go modules and the recommended approach for higher versions is to have a versioned sub directory. Therefore you will see there is now a v8 directory in the master branch. Would you be OK to revert the changes on these files and make them instead on the equivalent files in the v8 directory?

tyrannosaurus-becks commented:
@jcmturner sure, just did it! I pulled master into the branch, moved the changes to the `/v8` directory, and pushed them up again. I squashed the commits too, to make the PR cleaner. Let me know if you'd like any other changes.
